<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1163</width>
    <height>1137</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="QPushButton" name="ScanDevButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>80</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Scans for available Basler cameras&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>Scan
Devices</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ConnectButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>10</y>
      <width>80</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Connects to previously detected Basler cameras.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>Connect</string>
    </property>
   </widget>
   <widget class="QPushButton" name="RUNButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>210</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start grabbing from &lt;span style=&quot; text-decoration: underline;&quot;&gt;all &lt;/span&gt;cameras without recording.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {background-color:  rgb(0, 255, 0)}</string>
    </property>
    <property name="text">
     <string>RUN</string>
    </property>
   </widget>
   <widget class="QPushButton" name="STOPButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>210</y>
      <width>80</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Stop grabbing from cameras.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {color: white; background-color: gray;}</string>
    </property>
    <property name="text">
     <string>STOP</string>
    </property>
   </widget>
   <widget class="QPushButton" name="RECButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>210</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start recording videos from all cameras.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {background-color:  rgb(200, 0, 0)}</string>
    </property>
    <property name="text">
     <string>REC</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>80</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Frame rate</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="FrameRateSpin">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>100</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Desired framerate.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="suffix">
     <string> fps</string>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
   </widget>
   <widget class="QTextEdit" name="Devices_textEdit">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>161</width>
      <height>131</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;List of detected Basler cameras (SN).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="AutoExposeButton">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>70</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Run auto-exposure.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>AutoExpose</string>
    </property>
   </widget>
   <widget class="QPushButton" name="AutoGainButton">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>100</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Run auto-gain.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>AutoGain</string>
    </property>
   </widget>
   <widget class="QPushButton" name="WhiteBalanceButton">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>130</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Run auto white balance. (Color cams only).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>WhiteBalance</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="All_cams_checkBox">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>40</y>
      <width>75</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Whether to run auto-functions on &lt;span style=&quot; text-decoration: underline;&quot;&gt;all&lt;/span&gt; cams.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>all cams</string>
    </property>
   </widget>
   <widget class="QPushButton" name="FlipXButton">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>170</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flips image along X-Axis.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>Flip X</string>
    </property>
   </widget>
   <widget class="QPushButton" name="FlipYButton">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>200</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flips image along Y-Axis.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>Flip Y</string>
    </property>
   </widget>
   <widget class="MultiCameraViewer" name="MultiViewWidget" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>350</y>
      <width>1141</width>
      <height>771</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="CameraSettingsTab" name="CameraSettings" native="true">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>10</y>
      <width>301</width>
      <height>331</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QPushButton" name="Save_pathButton">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>40</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Opens a dialog to choose a folder where the videos will be saved in.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>Choose
Savepath</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="crf_spinBox">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>100</y>
      <width>71</width>
      <height>26</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Constant Rate Factor. Quality preset. Lower values better quality.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="maximum">
     <number>51</number>
    </property>
    <property name="value">
     <number>18</number>
    </property>
   </widget>
   <widget class="QComboBox" name="Codec_comboBox">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>70</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Codec used for encoding the video. Some might not be available on current system.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>70</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Codec</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>100</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>crf</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>40</y>
      <width>211</width>
      <height>17</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Video Encoding params</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="SessionIDlineEdit">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>180</y>
      <width>371</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Prefix of the filename e.g., session_id.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>160</y>
      <width>121</width>
      <height>17</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Video name prefix:</string>
    </property>
   </widget>
   <widget class="QLabel" name="Rec_status">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>71</width>
      <height>51</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Current status.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>130</y>
      <width>141</width>
      <height>21</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Desired framerate as a slider.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>160</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksAbove</enum>
    </property>
    <property name="tickInterval">
     <number>20</number>
    </property>
   </widget>
   <widget class="QLabel" name="SavePath_label">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>10</y>
      <width>321</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="recording_duration_label">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>210</y>
      <width>91</width>
      <height>17</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>0s</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ScreenshotButton">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>220</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start recording videos from all cameras.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {background-color:  rgb(200, 0, 0)}</string>
    </property>
    <property name="text">
     <string>Screenshot</string>
    </property>
   </widget>
   <widget class="QPushButton" name="markerAddButton">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>300</y>
      <width>61</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Add
Markers</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="markerClearButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>300</y>
      <width>61</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear
Markers</string>
    </property>
   </widget>
   <widget class="QDial" name="Rotation_dial">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>270</y>
      <width>50</width>
      <height>64</height>
     </rect>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
    <property name="sliderPosition">
     <number>180</number>
    </property>
   </widget>
   <widget class="QSlider" name="Grid_slider">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>300</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>150</number>
    </property>
    <property name="value">
     <number>40</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksAbove</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="GridButton">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>300</y>
      <width>61</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Show
Grid</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionsaveSettings">
   <property name="text">
    <string>SaveSettings</string>
   </property>
  </action>
  <action name="actionLoadSettings">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>LoadSettings</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MultiCameraViewer</class>
   <extends>QWidget</extends>
   <header location="global">ImageViewer</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CameraSettingsTab</class>
   <extends>QWidget</extends>
   <header>ImageViewer</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>FrameRateSpin</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>531</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>404</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>FrameRateSpin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>144</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
